### ISSUE TITLE
Invalid JSX error

### ISSUE THREAD

**Issue by gouldingken** [kevzettler/react-regl#35]
I'm trying to follow the basic triangle example, but I'm getting JSX errors. I've tried to see if I can change anything in the tsconfig file to work around this, but I can't see anything that you're doing different.

What am I missing?

```
'Triangle' cannot be used as a JSX component.
  Its return type 'IDregl<Regl>' is not a valid JSX element.
    Type 'IDregl<Regl>' is missing the following properties from type 'ReactElement<any, any>': type, props, key  TS2786

    36 |         <ReglFrame
    37 |             color={[0.40625, 0.94921, 0.996, 1]}>
  > 38 |             <Triangle />
       |              ^
    39 |         </ReglFrame>
    40 |     );
    41 | };

```

**Comment by kevzettler** [comment]
@gouldingken thanks for reporting. Will need more information. At first glance it seems like potential version conflict. Please share what versions of `react`, `react-regl` and `typescript` your working with.

**Comment by kevzettler** [comment]
Ok nevermind. I can reproduce on a fresh `create-react-app`. I'm not sure whats wrong yet. Very strange, the typescript compiler gets tripped up on some of the type overrides. Seems to work fine on the storybook though.

**Comment by gouldingken** [comment]
Thanks for taking the time to replicate. Yes, my environment was created recently with CRA (typescript template).

```
    "react": "^17.0.2",
    "react-dom": "^17.0.2",
    "react-regl": "^4.3.0",
    "react-scripts": "4.0.3",
    "typescript": "^4.1.2",
```

I can confirm that it also works for me when I run the examples in the storybook. But when I open the same examples in my IDE (WebStorm) I get the same typescript error I reported above ( type 'IDregl<Regl>' is not a valid JSX element etc.).

**Comment by kevzettler** [comment]
fixed in efbbb5de60536b1997400abd0b9466d230d15005
published in 4.3.1

thanks for reporting @gouldingken
